common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerTokenProvider.java

```diff
@@ -0,0 +1,77 @@
+            throws ClientException {
+        final String methodName = ":acquireTokenInteractively";
+
+        if (activity == null) {
+            throw new ClientException(ErrorStrings.EMPTY_ANDROID_CONTEXT);
+        }
+
+        final Bundle requestBundle = getBrokerOptions(request);
+        final Intent brokerIntent = MicrosoftAuthServiceHandler.getInstance().getIntentForInteractiveRequest(activity);
+        if (brokerIntent == null) {
+            ClientException exception = new ClientException(ErrorStrings.BROKER_APP_NOT_RESPONDING, "Received null intent from broker interactive request.");
+            Logger.error(TAG, "Received null intent from broker interactive request.", exception);
+            throw exception;
+        } else {
+            brokerIntent.putExtras(requestBundle);
+            brokerIntent.putExtra(AuthenticationConstants.Broker.BROKER_REQUEST,
+                    AuthenticationConstants.Broker.BROKER_REQUEST);
+        }
+
+        Logger.verbose(TAG + methodName, "Calling activity. " + "Pid:" + android.os.Process.myPid()
+                + " tid:" + android.os.Process.myTid() + "uid:" + android.os.Process.myUid());
+        activity.startActivityForResult(brokerIntent, AuthenticationConstants.UIRequest.BROWSER_FLOW);
+    }
+
+
+    private Bundle getBrokerOptions(final MicrosoftAuthorizationRequest request) {
+        Bundle brokerOptions = new Bundle();
+        // request needs to be parcelable to send across process
+        brokerOptions.putString(AuthenticationConstants.Broker.ACCOUNT_AUTHORITY, request.getAuthority().toString());
+        brokerOptions.putString(AuthenticationConstants.Broker.ACCOUNT_RESOURCE, request.getAuthorizationEndpoint());
+        brokerOptions.putString(AuthenticationConstants.Broker.ACCOUNT_REDIRECT, request.getRedirectUri());
+        brokerOptions.putString(AuthenticationConstants.Broker.ACCOUNT_CLIENTID_KEY, request.getClientId());
+        brokerOptions.putString(AuthenticationConstants.Broker.ACCOUNT_USERINFO_USERID, request.getLoginHint());
+        brokerOptions.putString(AuthenticationConstants.Broker.ACCOUNT_LOGIN_HINT, request.getLoginHint());
+        brokerOptions.putString(AuthenticationConstants.Broker.ACCOUNT_NAME, request.getLoginHint());
+        brokerOptions.putString(AuthenticationConstants.Broker.ACCOUNT_EXTRA_QUERY_PARAM, request.getExtraQueryParam());
+        brokerOptions.putString(AuthenticationConstants.Broker.AUTH_RESPONSE_TYPE, request.getResponseType());
+        brokerOptions.putString(AuthenticationConstants.Broker.AUTH_STATE, request.getState());
+        brokerOptions.putString(AuthenticationConstants.Broker.AUTH_SCOPE, request.getScope());
+        brokerOptions.putString(AuthenticationConstants.Broker.LIB_NAME, request.getLibraryName());
+        brokerOptions.putString(AuthenticationConstants.Broker.LIB_VERSION, request.getLibraryVersion());
```

Comment:
can you add the apps name to the bundle as well?

Reply:
done